godot/Shaders/outline2D_inner.shader

```diff
@@ -1,19 +1,20 @@
 shader_type canvas_item;
 
-uniform vec4 line_color : hint_color = vec4(1);
+uniform vec4 line_color : hint_color = vec4(1.0);
 uniform float line_thickness : hint_range(0, 10) = 1.0;
 
+const vec2 OFFSETS[8] = {
+	vec2(-1, -1), vec2(-1, 0), vec2(-1, 1), vec2(0, -1), vec2(0, 1), 
+	vec2(1, -1), vec2(1, 0), vec2(1, 1)
+};
+
 void fragment() {
 	vec2 size = TEXTURE_PIXEL_SIZE * line_thickness;
+	float outline = 1.0;
 	
-	float outline = texture(TEXTURE, UV + vec2(-size.x, 0)).a;
-	outline *= texture(TEXTURE, UV + vec2(0, size.y)).a;
-	outline *= texture(TEXTURE, UV + vec2(size.x, 0)).a;
-	outline *= texture(TEXTURE, UV + vec2(0, -size.y)).a;
-	outline *= texture(TEXTURE, UV + vec2(-size.x, size.y)).a;
-	outline *= texture(TEXTURE, UV + vec2(size.x, size.y)).a;
-	outline *= texture(TEXTURE, UV + vec2(-size.x, -size.y)).a;
-	outline *= texture(TEXTURE, UV + vec2(size.x, -size.y)).a;
+	for (int i = 0; i < OFFSETS.length(); i++) {
+		outline *= texture(TEXTURE, UV + size * OFFSETS[i]).a;
+	}
```

Comment:
Be sure to write down to complete the corresponding outline tutorial with this code refactoring. In this case, I'd keep the code as it is presented in the course, with all those explicit lines, and then show the shorter version with the loop I think.